src/main/clojure/clara/rules/dsl.clj

```diff
@@ -294,7 +294,46 @@
   ([lhs rhs properties]
      (parse-rule* lhs rhs properties &env)))
 
+;;; added to clojure.core in 1.9
+(defn- qualified-keyword?
+  "Return true if x is a keyword with a namespace"
+  [x] (and (keyword? x) (namespace x) true))
+
```

Comment:
Don't we need schema changes for [Rule](https://github.com/cerner/clara-rules/blob/0.17.0/src/main/clojure/clara/rules/schema.cljc#L66) and [Query](https://github.com/cerner/clara-rules/blob/0.17.0/src/main/clojure/clara/rules/schema.cljc#L66)?  It is interesting that we didn't fail on them - we probably aren't making appropriate assertions, which may be legitimate since we've seen schema checks be a perf problem in Clara before - but I'd rather have the schemas be correct in any case since they are also a form of documentation.

Reply:
Did you want the changes in this PR, or is this a separate issue?

Reply:
In this PR please since it is basically changing the documentation to reflect the changes already in the PR.  To be clear, I'm not saying that we need to increase schema validation etc. in this PR.  Just changing the schemas in the appropriate places from accepting a String to accepting either a String or keyword is sufficient IMO.

Reply:
Done. While I was in there I updated the use of `s/either` to `s/cond-pre` to get rid of the deprecation warning.